xonsh/tools.py

```diff
@@ -909,3 +909,50 @@ def check_for_partial_string(x):
         return (string_indices[-1], None, starting_quote[-1])
     else:
         return (string_indices[-2], string_indices[-1], starting_quote[-1])
+
+
+# expandvars is a modified version of os.path.expandvars from the Python 3.5.1
+# source code (root/Lib/posixpath.py, line 266)
+
+_varprog = None
+
+
+def expandvars(path):
```

Comment:
Well, one thing is that I think it is nice if this works with the `${..}` syntax (none of the `expandvars` in `os` will do that).

More important is the fallback to default values, which wouldn't work with `replace_env`.